### DISCUSSION TITLE
Is there anyway to speed up the compilation time?

### DISCUSSION THREAD

**Discussion by macohi** [FunkinCrew/Funkin#3899] [Q&A] [answered]
Im now a programmer on someones mod and im recompiling the game ALOT.
So I was wondering if there was a way to increase the speed of compiling.

**Comment by MrMadera** [comment]
i don't really think so... the only way to improve compilation time is by having a good cpu. hope there's another way

**Reply by THEkatinamicrowave** [reply]
big cpu

**Reply by THEkatinamicrowave** [reply]
also I think if you've already compiled once, all subsequent compiles will take significantly shorter since it's just adding .cpp stuff and assets.

**Reply by MrMadera** [reply]
> also I think if you've already compiled once, all subsequent compiles will take significantly shorter since it's just adding .cpp stuff and assets.

true! as far as I know, the first time you compile, haxe, every library and the game are compiled into .cpp files, so it takes a longer time. the following times, it will reduce the time significantly because it's only compiling the changes you make

**Comment by NotHyper-474** [comment]
You could compile to HTML5 which is very fast but doesn't seem to have mod support nor debug states (like the chart editor), or to HashLink which usually compiles in under 30 seconds for me but requires some [changes](https://github.com/NotHyper-474/Funkin-Legacy-Linux/commit/a3f577819546ce675897a9e29a9eabc4da2a67c9) to work and even compile, however it's very crash-prone and Polymod is really broken in it but it is good enough for me to do some quick testing.

**Reply by macohi** [reply]
i wish i could compile to Hashlink but they added discord rpc support

**Reply by macohi** [reply]
and a bunch of other stuff im pretty sure

**Reply by macohi** [reply]
which cant compile with hashlink

**Reply by NotHyper-474** [reply]
Discord RPC can be disabled by specifying `NO_FEATURE_DISCORD_RPC`, compiling to Debug or doing what my one of my changes did in the commit I linked, unless you mean you want Discord RPC support?

**Reply by MrMadera** [reply]
i've never tried compiling to hashlink before, does it actually make a difference compared to compiling to cpp??

**Reply by NotHyper-474** [reply]
> i've never tried compiling to hashlink before, does it actually make a difference compared to compiling to cpp??

Yep, despite it crashing a lot and having some bugs it's way quicker to compile (at least for me) than to C++ and it has very similar performance. And if I'm not mistaken even the devs use it.

I would only recommend it for quick testing though.

**Reply by MrMadera** [reply]
oh got it! i'll take a look later and test it out

**Reply by macohi** [reply]
> Discord RPC can be disabled by specifying `NO_FEATURE_DISCORD_RPC`, compiling to Debug or doing what my one of my changes did in the commit I linked, unless you mean you want Discord RPC support?

in windows builds, in hashlink idc

**Reply by NotHyper-474** [reply, marked as answer]
You said you were compiling a lot so I supposed that was to test your
changes. Even so, you can compile to HashLink for that, then when you're
done testing, for Windows/C++.

**Reply by macohi** [reply]
> You said you were compiling a lot so I supposed that was to test your changes. Even so, you can compile to HashLink for that, then when you're done testing, for Windows/C++.

yeah sounds about right
